File: Coding Challenge/Program 9/Program1/Address.cpp
#include "Address.h"

Address::Address(const std::wstring &sn, int hn, const std::wstring &an, const std::wstring &ct, const std::wstring &st, const std::wstring &cou)
{
	streetName = sn;
	houseNo = hn;
	aptNo = an;
	city = ct;
	state = st;
	country = cou;
}

void Address::display()
{
	std::wcout << houseNo << L" " << streetName << std::endl;
	if (!aptNo.isEmpty())
	{
		std::wcout << L"Apt #: " << aptNo << std::endl;
	}
	std::wcout << city << L", " << state << std::endl;
	std::wcout << country << std::endl;
}

File: Coding Challenge/Program 9/Address.java
public class Address{
    private String streetName;
    private int houseNo;
    private String aptNo;
    private String city;
    private String state;
    private String country;

    public Address(String sn, int hn, String an, String ct, String st, String cou){
        streetName = sn;
        houseNo = hn;
        aptNo = an;
        city=ct;
        state=st;
        country=cou;
    }
    public void display(){
        System.out.println(houseNo+" "+streetName);
        if(!aptNo.isEmpty()){
            System.out.println("Apt #: "+aptNo);
        }
        System.out.println(city+", "+state);
        System.out.println(country);
    }
}